runelite-client/src/main/java/net/runelite/client/plugins/praydose/PrayerDosePlugin.java

```diff
@@ -0,0 +1,283 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.praydose;
+
+import com.google.common.collect.Sets;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.api.Prayer;
+import net.runelite.api.Skill;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.inject.Inject;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+
+@PluginDescriptor(
+	name = "Prayer Dose"
+)
+public class PrayerDosePlugin extends Plugin
+{
+	private static final Set<Integer> SUPER_RESTORE_ITEM_IDS = Sets.newHashSet
+	(
+		ItemID.SUPER_RESTORE4, ItemID.SUPER_RESTORE3, ItemID.SUPER_RESTORE2, ItemID.SUPER_RESTORE1
+	);
+
+	private static final Set<Integer> PRAYER_ITEM_IDS = Sets.newHashSet
+	(
+		ItemID.PRAYER_POTION4, ItemID.PRAYER_POTION3, ItemID.PRAYER_POTION2, ItemID.PRAYER_POTION1
+	);
+
+	private static final Set<Integer> HOLY_EQUIPMENT_IDS = Sets.newHashSet
+	(
+		ItemID.PRAYER_CAPE, ItemID.PRAYER_CAPET, ItemID.PRAYER_CAPE_10643,
+		ItemID.MAX_CAPE, ItemID.MAX_CAPE_13282, ItemID.MAX_CAPE_13342,
+		ItemID.HOLY_WRENCH, ItemID.RING_OF_THE_GODS, ItemID.RING_OF_THE_GODS_I
+	);
+
+	@Getter
+	private boolean holdingPrayerPotion = false;
+
+	@Getter
+	private boolean holdingSuperRestore = false;
+
+	@Getter
+	private boolean holyWrench = false;
+
+	@Getter
+	private int prayerPotionPointsRestored = 7;
+
+	@Getter
+	private int superRestorePointsRestored = 8;
+
+	@Inject
+	private PrayerDoseOverlay overlay;
+
+	@Inject
+	private PrayerDoseConfig config;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Provides
+	PrayerDoseConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(PrayerDoseConfig.class);
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	@Subscribe
+	public void onTick(GameTick tick)
+	{
+		overlay.onTick();
+
+		this.holdingPrayerPotion = false;
+		this.holdingSuperRestore = false;
+		this.holyWrench = false;
+
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		this.checkInventory();
```

Comment:
Instead of checkEquipment and checkInventory each game tick, just use ItemContainerChanged event, basically, remove this entire gameTick hook and use proper events